### PR TITLE
[GLUTEN-3072][CH] Optimization for `get_json_object`

### PR DESCRIPTION
**Pull request by lgbo-ustc**
## What changes were proposed in this pull request?

(Please fill in changes proposed in this fix)

Fixes: #3072

In #2410, we make the intermediate data shared among multiple `get_json_object`s, but found that
- the intermediate data is large, need a lot of memory and cause OOM easly.
- cache miss is more serious, it only bring +10% speedup at final.

In this PR, we take another way, rewrite the projection rel. 
- collect all required fields on the same json string, and construct a new function, `flattenJSONStringOnRequired(json_str, required_fields)`, `flattenJSONStringOnRequired` will flatten a json string into a named tuple by the required fields.
- replace all `get_json_object`'s 1st argument with `flattenJSONStringOnRequired(json_str, required_fields)`.
- `GetJSONObjectParser` will check whether `get_json_object`'s 1st argument has been rewritten. If the argument has been rewritten, `get_json_object` will be explained to `flattenJSONStringOnRequired` + `tumpleElement`. Otherwise keep it as before.

## How was this patch tested?

(Please explain how this patch was tested. E.g. unit tests, integration tests, manual tests)

unit tests

(If this patch involves UI changes, please attach a screenshot; otherwise, remove this)

### CODE REVIEW

**Comment by lgbo-ustc** [comment]
test result.

| repeated functions   | before | after |
| -------- | ------- | ------- |
|7  | 27s   | 17s|
|2 | 9.4s     | 8.9s |
| 1    | 6.6s |  6.7s   |

File: cpp-ch/local-engine/Storages/SubstraitSource/ORCFormatFile.cpp [hunk 1]
```diff
@@ -208,7 +209,15 @@ FormatFile::InputFormatPtr ORCFormatFile::createInputFormat(const DB::Block & he
         std::back_inserter(skip_stripe_indices));
 
     format_settings.orc.skip_stripes = std::unordered_set<int>(skip_stripe_indices.begin(), skip_stripe_indices.end());
-    auto input_format = std::make_shared<DB::ORCBlockInputFormat>(*file_format->read_buffer, header, format_settings);
+    DB::InputFormatPtr input_format = nullptr;
+    if (format_settings.orc.use_fast_decoder)
```
Comment (taiyang-li) [inline cpp-ch/local-engine/Storages/SubstraitSource/ORCFormatFile.cpp:213]: format factory中已经包含了根据开关选择的逻辑，是否可通过FormatFactory::getInput 接口复用

**Review by taiyang-li — CHANGES_REQUESTED** [review]

File: cpp-ch/local-engine/Storages/SubstraitSource/ORCFormatFile.cpp (hunk 1 above)
Comment (lgbo-ustc) [reply cpp-ch/local-engine/Storages/SubstraitSource/ORCFormatFile.cpp:213]: 有问题。无法使用factory的接口

Comment (lgbo-ustc) [reply cpp-ch/local-engine/Storages/SubstraitSource/ORCFormatFile.cpp:213]: 先去掉这个带动了，这个PR对内存的需求已经变小了

**Comment by lwz9103** [comment]
Run Gluten Clickhouse CI

**Review by liuneng1994 — APPROVED** [review]
LGTM

**Pull request merged**